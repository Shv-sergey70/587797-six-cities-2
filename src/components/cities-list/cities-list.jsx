import React from 'react';
import PropTypes from 'prop-types';
import City from "../city/city";
import {connect} from "react-redux";
import cityPropTypes from '../../prop-types/city';

const CitiesList = (props) => {
  const {
    cities
  } = props;

  return <ul className="locations__list tabs__list">
    {cities.map((city, i) => <City
      key = {`city-${i}`}
      city = {city}
    />)}
  </ul>;
};

CitiesList.propTypes = {
  cities: PropTypes.arrayOf(cityPropTypes)
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  cities: state.cities
});

export {CitiesList};

export default connect(mapStateToProps)(CitiesList);
