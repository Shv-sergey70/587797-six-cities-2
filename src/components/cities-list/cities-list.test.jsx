import React from 'react';
import renderer from 'react-test-renderer';
import {CitiesList} from "./cities-list";

jest.mock(`../city/city.jsx`, () => `City`);

const citiesData = [
  {
    name: `Paris`,
    location: {
      latitude: 11,
      longitude: 22,
      zoom: 10
    }
  },
  {
    name: `Brussels`,
    location: {
      latitude: 11,
      longitude: 22,
      zoom: 10
    }
  },
  {
    name: `Amsterdam`,
    location: {
      latitude: 11,
      longitude: 22,
      zoom: 10
    }
  }
];

it(`Cities list correctly renders`, () => {
  const tree = renderer.create(
      <CitiesList
        cities = {citiesData}
      />).toJSON();

  expect(tree).toMatchSnapshot();
});
