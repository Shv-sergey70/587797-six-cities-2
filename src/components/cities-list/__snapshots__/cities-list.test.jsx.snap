// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities list correctly renders 1`] = `
<ul
  className="locations__list tabs__list"
>
  <City
    city={
      Object {
        "location": Object {
          "latitude": 11,
          "longitude": 22,
          "zoom": 10,
        },
        "name": "Paris",
      }
    }
  />
  <City
    city={
      Object {
        "location": Object {
          "latitude": 11,
          "longitude": 22,
          "zoom": 10,
        },
        "name": "Brussels",
      }
    }
  />
  <City
    city={
      Object {
        "location": Object {
          "latitude": 11,
          "longitude": 22,
          "zoom": 10,
        },
        "name": "Amsterdam",
      }
    }
  />
</ul>
`;
